src/dotnet/commands/dotnet-test/BaseDotnetTestRunner.cs

```diff
@@ -36,11 +36,8 @@ private int DoBuildTestProject(DotnetTestParams dotnetTestParams)
                 $"{dotnetTestParams.ProjectPath}"
             };
 
-            if (dotnetTestParams.Framework != null)
-            {
-                strings.Add("--framework");
-                strings.Add($"{dotnetTestParams.Framework}");
-            }
+            strings.Add("--framework");
```

Comment:
This avoids rebuilding the project for all TFM for each TFM that tests are being run for.